app/assets/javascripts/host_edit.js

```diff
@@ -158,10 +158,13 @@ function submit_with_all_params() {
     data: serializeForm(),
     success: function(response) {
       $('#host-progress').hide();
-      $('#content').replaceWith($('#content', response));
-      $(document.body).trigger('ContentLoad');
-      if ($("[data-history-url]").exists()) {
-        history.pushState({}, 'Host show', $("[data-history-url]").data('history-url'));
+      var host_url = response.match(new RegExp("data-history-url='" + "(.*)" + "'"));
```

Comment:
```suggestion
      var host_url = response.match(new RegExp("data-history-url='(.*)'"));
```

But if we're doing a full page reload, why not just use the redirect the controller replies with instead of this hack that was needed to update the current url without reloading the page? when the host finishes orchestration the server replies with 302 to the new show page.